ci/unit_tests/functions_test/main_test.py

```diff
@@ -0,0 +1,268 @@
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'CRITICAL Cannot decode json from input payload from file', # error message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_testSinglePayloadsOutNonExistingOutput(self):
+        ''' Tests if the single test where input and expected output is specified in another file - by relative path to that file but expected output path is wrong (all other params are given from command line) '''
+        testArgs = [self.testSinglePayloadsOutNonExistingOutputJsonPath, self.noJsonPath] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'CRITICAL Cannot open expected output payload from file: /some/random/path', # error message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_testSinglePayloadsOutNoJsonOutput(self):
+        ''' Tests if the single test where input and expected output is specified in another file - by relative path to that file but expected output is invalid json (all other params are given from command line) '''
+        testArgs = [self.testSinglePayloadsOutNoJsonOutputJsonPath, self.noJsonPath] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'CRITICAL Cannot decode json from expected output payload from file', # error message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_testSingleAllInFileBadType(self):
+        ''' Tests if bad test type was provided '''
+        testArgs = [self.testSingleAllInFileBadTypeJsonPath, self.noJsonPath] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'CRITICAL Error while evaluation, unknown test type: BAD TYPE', # error message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_testSingleAllInFileOverridePackage(self):
+        ''' Tests if bad package was provided in test it self, the main reason is test that override works '''
+        testArgs = [self.testSingleAllInFileOverridePackageJsonPath, self.noJsonPath] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'actions/BAD_PACKAGE/echo', # log message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_testSingleAllInFileOverrideFunction(self):
+        ''' Tests if bad package was provided in test it self, the main reason is test that override works '''
+        testArgs = [self.testSingleAllInFileOverrideFunctionJsonPath, self.noJsonPath] + self.functionsTestArgs
+        self.t_exitCodeAndLogMessage(
+            1, # exit code
+            'actions/utils/BAD_FUNCTION', # log message substring
+            [testArgs] # params (*args, **kwargs)
+        )
+
+    def test_testSingleMulti(self):
```

Comment:
I think that correct name should be `test_multi`.

Reply:
I'll improve it.